public/scripts/report.js

```diff
@@ -1,24 +1,107 @@
 (function(){
 
-  // GET LOCATION
-  var locationButton = document.getElementById('getLocationButton');
-
-  function getLocation() {
-    if (navigator.geolocation) {
-      navigator.geolocation.getCurrentPosition(showPosition);
-    } else {
-      // non-supprt error handling
-      console.log('Geolocation is not supported by this browser.');
+  function getLocationListener(){
+
+    var locationButton = document.getElementById('getLocationButton');
+    locationButton.addEventListener('click', getLocation);
+
+    // GET USER'S LOCATION
+    function getLocation() {
+
+      locationButton.innerHTML = 'Getting your location...';
+      locationButton.classList.add('getting-location');
+
+      if (navigator.geolocation) {
+
+        navigator.geolocation.getCurrentPosition(storePosition, function(err){
+
+          console.log('error :',err);
+
+          locationButton.innerHTML = 'Error Getting Location';
+
+          locationButton.classList.remove('getting-location');
+          locationButton.classList.add('error-getting-location');
+
+        });
+
+      } else {
+        // non-supprt error handling
+        console.log('Geolocation is not supported by this browser.');
+      }
+    };
+
+    // store user's location in button attribute
+    function storePosition(position) {
+
+      var coords = position.coords.latitude+","+position.coords.longitude;
+
+      locationButton.setAttribute("value", coords);
+
+      locationButton.innerHTML = 'Using current location';
+
+      locationButton.classList.remove('getting-location');
+
+      locationButton.classList.add('using-location');
+
+    };
+
+  }
+
+    // INSERT SUBMISSION TO DB (SUBMIT HANDLER)
+    function submitFormListener(){
+
+      var reportForm = document.getElementById('reportForm');
+
+      reportForm.addEventListener('submit', submitHandler)
+
+      function submitHandler(event){
```

Comment:
no need for this function - you can just call extractFormData immediately, and move the prevent default inside that